### ISSUE TITLE
"You can't vote for your own comment" in user comments sidebar

### ISSUE THREAD

**Issue by Reeywhaar** [umputun/remark#252]
Noticed long ago, but it's still there, so, all upvote/downvote buttons in user's comments sidebar show "you can't vote for your own comment" hover title, though I'm not even logged in.
![screenshot 2019-01-17 at 00 17 18](https://user-images.githubusercontent.com/884649/51279195-b9ffbf80-19ed-11e9-901d-d95d111320b9.png).

There shound be change to either change title, or hide upvote/downvote completely.

**Comment by Reeywhaar** [comment]
Found out that sending request to get own user comments results in missing `comments` field in response:
```
Get own comments:

Request:
GET /api/v1/comments?user=dev_user&limit=10&site=remark HTTP/1.1
X-XSRF-TOKEN: ▊▊▊▊▊▊▊▊
Cookie: JWT=▊▊▊▊▊▊▊; XSRF-TOKEN=▊▊▊▊▊

Response:
{"count":21}
```

```
Get someone's comments

Request:
GET /api/v1/comments?user=jimbo&limit=10&site=remark HTTP/1.1
X-XSRF-TOKEN: ▊▊▊▊▊▊▊▊
Cookie: JWT=▊▊▊▊▊▊▊▊; XSRF-TOKEN=▊▊▊▊▊▊▊▊

Response:
{"comments":[▊▊▊▊▊▊▊▊],"count":1}
```

```
Get comments while logged out

Request:
GET /api/v1/comments?user=dev_user&limit=10&site=remark HTTP/1.1

Response:
{"comments":[▊▊▊▊▊▊▊▊],"count":21}

```

Am I missing something?

**Comment by umputun** [comment]
this is odd. There is no difference which `user` requested, this is an open api and has nothing to do with the authorized (or not) user. 

I'll try to reproduce it with some unit test to figure what's going on.

Do you see any error messages in the log (console)?

**Comment by Reeywhaar** [comment]
No, both browser's and backend's logs are clear
```
remark42-dev | 2019/01/16 18:18:26.938 [INFO]  {logger/logger.go:107} GET - /api/v1/user?site=remark - 7a5001c67cde - 200 (139) - 59.947µs
remark42-dev | 2019/01/16 18:18:26.940 [DEBUG] {api/rest_public.go:183} get comments for userID dev_user, remark
remark42-dev | 2019/01/16 18:18:26.940 [INFO]  {logger/logger.go:107} GET - /api/v1/comments?user=dev_user&limit=10&site=remark - 7a5001c67cde - 200 (13)- 271.035µs
```

**Comment by Reeywhaar** [comment]
Oh, really strange set of circumstances. It has nothing to do with "logged in/logged out". The real reason must be is that is there was a lot of comments by "dev user" that I have deleted, so "/api/v1/comments?user=dev_user" must be works like "get last 10 comments and then filter out deleted", so result gets empty.

**Comment by Reeywhaar** [comment]
Questions:
* Should voting be allowed in last comments view? Current version has voting disabled, but it's more a bug than decision.
* Should admin have comment's controls in last comments view? Like:
![screenshot 2019-01-17 at 03 51 14](https://user-images.githubusercontent.com/884649/51287973-29d07300-1a0b-11e9-80e8-211dc75f4879.png)

**Comment by umputun** [comment]
> Should voting be allowed in last comments view? 

I don't think so. We don't want to provoke users to mass up/down voting other users.

> Should admin have comment's controls in last comments view? 

Would be nice I guess. Not that important, but could be useful

**Comment by Reeywhaar** [comment]
Edited #254 

Think "pin, block" controls are useless, and should be removed, am I right?

**Comment by umputun** [comment]
> must be works like "get last 10 comments and then filter out deleted", so result gets empty.

This is correct. Actually, such logic implemented for boltdb engine specifically as a part of optimization. List of user's comments saved as references and the moment it retrieves the list, the engine has no clue if a comment deleted or not. 

I don't consider such behavior as a bug, but results can be confusing. Hopefully, it can be addressed as a part of #155 change.

**Comment by umputun** [comment]
> Think "pin, block" controls are useless, and should be removed, am I right?

pin is useless for sure in that list. Block can be useful.
